File: src/main/java/com/matag/game/config/ConfigService.java
package com.matag.game.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import lombok.Getter;

@Component
@Getter
public class ConfigService {
  @Value("${matag.admin.path}")
  private String matagAdminPath;

  @Value("${matag.admin.internal.url}")
  private String matagAdminInternalUrl;

  @Value("${matag.admin.password}")
  private String adminPassword;
}

File: src/main/kotlin/com/matag/game/config/ConfigService.kt
package com.matag.game.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
open class ConfigService(
    @param:Value("\${matag.admin.path}")
    val matagAdminPath: String,

    @param:Value("\${matag.admin.internal.url}")
    val matagAdminInternalUrl: String,

    @param:Value("\${matag.admin.password}")
    val adminPassword: String
)
